dm/master/shardddl/optimist.go

```diff
@@ -500,8 +516,7 @@ func (o *Optimist) handleLock(info optimism.Info, tts []optimism.TargetTable, sk
 
 	lock := o.lk.FindLock(lockID)
 	if lock == nil {
-		// this aways means others remove the lock concurrently when resolved ddl.
-		// simply try again.
+		// should not happen
```

Comment:
can we remove `return o.handleLock(info, tts, skipDone)` in L 521 now or report an error?